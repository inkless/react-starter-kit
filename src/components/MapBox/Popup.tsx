import React from 'react';
import { Popup } from 'react-map-gl';

import { theme } from '@src/theme';
import Text from '@src/components/base/Typography/Text';

export type PopupInfo = {
  longitude: number;
  latitude: number;
  title: string;
  subtitle: string;
};

export type Props = {
  popupInfo: PopupInfo | null;
  onClose?: () => void;
};

/**
 * Render a Popup in Map
 * When props.popup is null, return null
 *
 * @param props.popupInfo
 * @param props.onClose
 */
export function MapPopup({ popupInfo, onClose }: Props) {
  if (!popupInfo) {
    return null;
  }
  return (
    <Popup
      tipSize={5}
      anchor="top"
      longitude={popupInfo.longitude}
      latitude={popupInfo.latitude}
      closeOnClick={false}
      onClose={onClose}
    >
      <>
        <Text fontSize={3} color={theme.colors.grays[9]} fontWeight="bold" textAlign="center">
          {popupInfo.title}
        </Text>
        <Text fontSize={3} color={theme.colors.grays[7]} p={0}>
          {popupInfo.subtitle}
        </Text>
      </>
    </Popup>
  );
}

export default MapPopup;
