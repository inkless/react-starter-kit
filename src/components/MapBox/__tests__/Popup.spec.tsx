import React from 'react';
import { shallow } from 'enzyme';
import { Popup as MapGLPopup } from 'react-map-gl';

import Popup from '../Popup';

describe('Component Popup', () => {
  it('should not render Popup if info is null', () => {
    const component = shallow(<Popup popupInfo={null} />);
    expect(component).toMatchSnapshot();
  });

  it('should render MapGL Popup if info is provided', () => {
    const popupInfo = {
      longitude: -100,
      latitude: 37,
      title: 'Title',
      subtitle: 'Sub Title',
    };

    const component = shallow(<Popup popupInfo={popupInfo} />);
    expect(component).toMatchSnapshot();
  });

  it('should call onClose when popup is closed', () => {
    const popupInfo = {
      longitude: -100,
      latitude: 37,
      title: 'Title',
      subtitle: 'Sub Title',
    };

    const onCloseSpy = jest.fn();
    const component = shallow(<Popup popupInfo={popupInfo} onClose={onCloseSpy} />);
    const mapGlPopup = component.find(MapGLPopup);
    expect(mapGlPopup).not.toBeNull();
    const props = mapGlPopup.props();
    props.onClose && props.onClose();
    expect(onCloseSpy).toHaveBeenCalled();
  });
});
