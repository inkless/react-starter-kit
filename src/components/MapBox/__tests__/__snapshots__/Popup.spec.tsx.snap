// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Popup should not render Popup if info is null 1`] = `""`;

exports[`Component Popup should render MapGL Popup if info is provided 1`] = `
<Popup
  altitude={0}
  anchor="top"
  captureClick={true}
  captureDoubleClick={true}
  captureDrag={true}
  captureScroll={false}
  className=""
  closeButton={true}
  closeOnClick={false}
  dynamicPosition={true}
  latitude={37}
  longitude={-100}
  offsetLeft={0}
  offsetTop={0}
  onClose={[Function]}
  sortByDepth={false}
  tipSize={5}
>
  <Styled(styled.div)
    color="#484848"
    fontSize={3}
    fontWeight="bold"
    textAlign="center"
  >
    Title
  </Styled(styled.div)>
  <Styled(styled.div)
    color="#7a7a7a"
    fontSize={3}
    p={0}
  >
    Sub Title
  </Styled(styled.div)>
</Popup>
`;
